stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/extensions/ChatClient.kt

```diff
@@ -33,4 +37,18 @@ public val ChatClient.globalState: GlobalState
     get() = GlobalMutableState.getOrCreate()
 
 @InternalStreamChatApi
-public fun ChatClient.asReferenced(): ChatClientReferenceAdapter = ChatClientReferenceAdapter(this)
+internal fun ChatClient.requestsAsState(): ChatClientStateCalls = ChatClientStateCalls(this, state)
+
+public fun ChatClient.queryChannelsAsState(request: QueryChannelsRequest): QueryChannelsState {
+    return requestsAsState().queryChannels(request)
+}
+
+public fun ChatClient.watchChannelAsState(cid: String, limit: Int = DEFAULT_MESSAGE_LIMIT): ChannelState {
```

Comment:
What do you think about removing the default `limit` parameter here and in `getRepliesAsState `? There are cases when we don't need any messages (for example `MessageInputView`) and it's easy forgetting to change the default limit. Making this parameter required would force everyone to think if they really need any messages